bb_wrapper/models/pagamentos.py

```diff
@@ -74,25 +76,35 @@ def _set_data(cls, values):
         from ..services.pix import PixService
 
         key = values.get("chave")
-
         key_type = PixService().identify_key_type(key)
-        values["formaIdentificacao"] = key_type
+        values["formaIdentificacao"] = key_type.value
 
         if key_type == TipoChavePIX.telefone:
-            values["dddTelefone"] = int(key[:2])
-            values["telefone"] = int(key[2:])
+            values["dddTelefone"] = key[:2]
+            values["telefone"] = key[2:]
         elif key_type == TipoChavePIX.email:
+            PixService().verify_email(key)
             values["email"] = key
         elif key_type == TipoChavePIX.uuid:
             values["identificacaoAleatoria"] = key
         elif key_type == TipoChavePIX.documento:
-            key_value = cpfcnpj.clear_punctuation(key)
-            if len(key_value) == 1:
-                values["cpf"] = int(key_value)
-            else:
-                values["cnpj"] = int(key_value)
+            cls.verifica_documento(key, values)
+
+        documento = values.pop("documento")
```

Comment:
Seria bom colocar o fallback do pop como None para que não estoure um KeyError se por algum motivo não for adicionado a chave 'documento'

```suggestion
        documento = values.pop("documento", None)
```